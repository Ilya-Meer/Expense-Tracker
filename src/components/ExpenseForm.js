import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addExpense } from '../actions/expenseActions'


class ExpenseForm extends Component {
  constructor(props) {
    super(props)

    this.state = {
      desc: '', 
      amount: 0,
      note: ''
    }

    this.handleDescription = this.handleDescription.bind(this);
    this.handleAmount = this.handleAmount.bind(this);
    this.handleNote = this.handleNote.bind(this);
  }

  handleDescription(e) {
    const desc = e.target.value
    this.setState(() => {
      return {
        ...this.state, 
        desc
      }
    })
  }
  handleAmount(e) {
    const amount = e.target.value
    this.setState(() => {
      return {
        ...this.state, 
        amount
      }
    })
  }
  handleNote(e) {
    const note = e.target.value
    this.setState(() => {
      return {
        ...this.state, 
        note
      }
    })
  }

  render() {
    return (
      <form onSubmit={(e) => {
          e.preventDefault();
          this.props.dispatch(addExpense(this.state))
        }}>
        <input autoFocus type="text" name="desc" placeholder="description" value={this.state.desc} onChange={this.handleDescription}/>
        <input type="number" name="amount" placeholder="amount" value={this.state.amount} onChange={this.handleAmount}/>
        <textarea name="note" id="note" cols="30" rows="10" placeholder="Add a note for your expense" value={this.state.note} onChange={this.handleNote}></textarea>
        <input type="submit" value="Submit"/>
      </form>
    )
  }
}


export default connect()(ExpenseForm);