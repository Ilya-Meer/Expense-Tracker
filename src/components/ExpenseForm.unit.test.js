import React from 'react';
import { shallow } from 'enzyme';
import ExpenseForm from './ExpenseForm';
import moment from './__mocks__/moment'

describe('Expense Form', () => {
  it('should render the Expense Form', () => {
    const wrapper = shallow(<ExpenseForm />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should render the Expense Form with seed data', () => {
    const expense = {
      id: '3',
      desc: 'Rent', 
      note: 'The rent was not cheap either', 
      amount: '2000000', 
      createdAt: moment(0)
    };

    const wrapper = shallow(<ExpenseForm expense={expense}/>);
    expect(wrapper).toMatchSnapshot();
  })

  it('should render an error message when the form is submitted without description or amount', () => {
    const wrapper = shallow(<ExpenseForm />);
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });
    expect(wrapper.state().error).toBe('Please make sure there is a valid description and amount!')
    expect(wrapper).toMatchSnapshot();
  })

  it('should set description on input change', () => {
    const wrapper = shallow(<ExpenseForm />);
    const input = wrapper.find('input').at(0).simulate('change', {
      target: {
        value: 'New Description'
      }
    });
    expect(wrapper.state().desc).toBe('New Description')
    expect(wrapper).toMatchSnapshot();
  })

  it('should set note on textarea change', () => {
    const wrapper = shallow(<ExpenseForm />);
    const textarea = wrapper.find('textarea').simulate('change', {
      target: {
        value: 'New Note'
      }
    });
    expect(wrapper.state().note).toBe('New Note')
    expect(wrapper).toMatchSnapshot();
  })

  it('should not set amount when given invalid input', () => {
    const wrapper = shallow(<ExpenseForm />);
    const textarea = wrapper.find('input').at(1).simulate('change', {
      target: {
        value: '100.10000'
      }
    });
    expect(wrapper.state().amount).toBe('')
    expect(wrapper).toMatchSnapshot();
  })

  it('should set amount when given valid input', () => {
    const wrapper = shallow(<ExpenseForm />);
    const textarea = wrapper.find('input').at(1).simulate('change', {
      target: {
        value: '100.00'
      }
    });
    expect(wrapper.state().amount).toBe('100.00')
    expect(wrapper).toMatchSnapshot();
  })

  it('should call the onSubmit prop when valid form is submitted', () => {
    const expense = {
      id: '3',
      desc: 'Rent', 
      note: 'The rent was not cheap either', 
      amount: '2000000', 
      createdAt: moment(0)
    };
    const onSubmitSpy = jest.fn();

    const wrapper = shallow(<ExpenseForm expense={expense} onSubmit={onSubmitSpy}/>);
    wrapper.find('form').simulate('submit', {
      preventDefault: () => {}
    });
    expect(onSubmitSpy).toHaveBeenCalledWith({
      desc: 'Rent', 
      note: 'The rent was not cheap either', 
      amount: 2000000, 
      createdAt: moment(0)
    });  
  })

  it('should set date on date change', () => {
    const now = moment()
    const wrapper = shallow(<ExpenseForm />);
    wrapper.find('withStyles(SingleDatePicker)').prop('onDateChange')(now);
    expect(wrapper.state('createdAt')).toEqual(now);
  })

  it('should set calendar focus on focus change', () => {
    const focused = true;
    const wrapper = shallow(<ExpenseForm />);
    wrapper.find('withStyles(SingleDatePicker)').prop('onFocusChange')({focused});
    expect(wrapper.state('calendarFocused')).toEqual(focused);
  })

})


