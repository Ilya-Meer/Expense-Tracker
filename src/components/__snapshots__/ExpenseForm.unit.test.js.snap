// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense Form should not set amount when given invalid input 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <p />
  <styled.input
    autoFocus={true}
    name="desc"
    onChange={[Function]}
    placeholder="description"
    type="text"
    value=""
  />
  <styled.input
    name="amount"
    onChange={[Function]}
    placeholder="amount"
    type="text"
    value=""
  />
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disabled={false}
    displayFormat="MMMM Do, YYYY"
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month",
        "jumpToPrevMonth": "Move backward to switch to the previous month",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
        "moveFocusByOneMonth": "Switch months",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
        "openThisPanel": "Open this panel",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field",
        "selectFocusedDate": "Select the date in focus",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    renderCalendarInfo={null}
    renderDay={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    cols="30"
    id="note"
    name="note"
    onChange={[Function]}
    placeholder="Add a note for your expense"
    rows="10"
    value=""
  />
  <styled.button
    type="submit"
  >
    <Component />
    Submit
  </styled.button>
</styled.form>
`;

exports[`Expense Form should render an error message when the form is submitted without description or amount 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <p>
    Please make sure there is a valid description and amount!
  </p>
  <styled.input
    autoFocus={true}
    name="desc"
    onChange={[Function]}
    placeholder="description"
    type="text"
    value=""
  />
  <styled.input
    name="amount"
    onChange={[Function]}
    placeholder="amount"
    type="text"
    value=""
  />
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disabled={false}
    displayFormat="MMMM Do, YYYY"
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month",
        "jumpToPrevMonth": "Move backward to switch to the previous month",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
        "moveFocusByOneMonth": "Switch months",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
        "openThisPanel": "Open this panel",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field",
        "selectFocusedDate": "Select the date in focus",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    renderCalendarInfo={null}
    renderDay={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    cols="30"
    id="note"
    name="note"
    onChange={[Function]}
    placeholder="Add a note for your expense"
    rows="10"
    value=""
  />
  <styled.button
    type="submit"
  >
    <Component />
    Submit
  </styled.button>
</styled.form>
`;

exports[`Expense Form should render the Expense Form 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <p />
  <styled.input
    autoFocus={true}
    name="desc"
    onChange={[Function]}
    placeholder="description"
    type="text"
    value=""
  />
  <styled.input
    name="amount"
    onChange={[Function]}
    placeholder="amount"
    type="text"
    value=""
  />
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disabled={false}
    displayFormat="MMMM Do, YYYY"
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month",
        "jumpToPrevMonth": "Move backward to switch to the previous month",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
        "moveFocusByOneMonth": "Switch months",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
        "openThisPanel": "Open this panel",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field",
        "selectFocusedDate": "Select the date in focus",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    renderCalendarInfo={null}
    renderDay={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    cols="30"
    id="note"
    name="note"
    onChange={[Function]}
    placeholder="Add a note for your expense"
    rows="10"
    value=""
  />
  <styled.button
    type="submit"
  >
    <Component />
    Submit
  </styled.button>
</styled.form>
`;

exports[`Expense Form should render the Expense Form with seed data 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <p />
  <styled.input
    autoFocus={true}
    name="desc"
    onChange={[Function]}
    placeholder="description"
    type="text"
    value="Rent"
  />
  <styled.input
    name="amount"
    onChange={[Function]}
    placeholder="amount"
    type="text"
    value="20000"
  />
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disabled={false}
    displayFormat="MMMM Do, YYYY"
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month",
        "jumpToPrevMonth": "Move backward to switch to the previous month",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
        "moveFocusByOneMonth": "Switch months",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
        "openThisPanel": "Open this panel",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field",
        "selectFocusedDate": "Select the date in focus",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    renderCalendarInfo={null}
    renderDay={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    cols="30"
    id="note"
    name="note"
    onChange={[Function]}
    placeholder="Add a note for your expense"
    rows="10"
    value="The rent was not cheap either"
  />
  <styled.button
    type="submit"
  >
    <Component />
    Submit
  </styled.button>
</styled.form>
`;

exports[`Expense Form should set amount when given valid input 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <p />
  <styled.input
    autoFocus={true}
    name="desc"
    onChange={[Function]}
    placeholder="description"
    type="text"
    value=""
  />
  <styled.input
    name="amount"
    onChange={[Function]}
    placeholder="amount"
    type="text"
    value="100.00"
  />
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disabled={false}
    displayFormat="MMMM Do, YYYY"
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month",
        "jumpToPrevMonth": "Move backward to switch to the previous month",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
        "moveFocusByOneMonth": "Switch months",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
        "openThisPanel": "Open this panel",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field",
        "selectFocusedDate": "Select the date in focus",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    renderCalendarInfo={null}
    renderDay={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    cols="30"
    id="note"
    name="note"
    onChange={[Function]}
    placeholder="Add a note for your expense"
    rows="10"
    value=""
  />
  <styled.button
    type="submit"
  >
    <Component />
    Submit
  </styled.button>
</styled.form>
`;

exports[`Expense Form should set description on input change 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <p />
  <styled.input
    autoFocus={true}
    name="desc"
    onChange={[Function]}
    placeholder="description"
    type="text"
    value="New Description"
  />
  <styled.input
    name="amount"
    onChange={[Function]}
    placeholder="amount"
    type="text"
    value=""
  />
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disabled={false}
    displayFormat="MMMM Do, YYYY"
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month",
        "jumpToPrevMonth": "Move backward to switch to the previous month",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
        "moveFocusByOneMonth": "Switch months",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
        "openThisPanel": "Open this panel",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field",
        "selectFocusedDate": "Select the date in focus",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    renderCalendarInfo={null}
    renderDay={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    cols="30"
    id="note"
    name="note"
    onChange={[Function]}
    placeholder="Add a note for your expense"
    rows="10"
    value=""
  />
  <styled.button
    type="submit"
  >
    <Component />
    Submit
  </styled.button>
</styled.form>
`;

exports[`Expense Form should set note on textarea change 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <p />
  <styled.input
    autoFocus={true}
    name="desc"
    onChange={[Function]}
    placeholder="description"
    type="text"
    value=""
  />
  <styled.input
    name="amount"
    onChange={[Function]}
    placeholder="amount"
    type="text"
    value=""
  />
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    daySize={39}
    disabled={false}
    displayFormat="MMMM Do, YYYY"
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    id="date"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPrev={null}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month",
        "jumpToPrevMonth": "Move backward to switch to the previous month",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
        "moveFocusByOneMonth": "Switch months",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
        "openThisPanel": "Open this panel",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field",
        "selectFocusedDate": "Select the date in focus",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    renderCalendarInfo={null}
    renderDay={null}
    renderMonth={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <textarea
    cols="30"
    id="note"
    name="note"
    onChange={[Function]}
    placeholder="Add a note for your expense"
    rows="10"
    value="New Note"
  />
  <styled.button
    type="submit"
  >
    <Component />
    Submit
  </styled.button>
</styled.form>
`;
